src/functions-and-arrays.js

```diff
@@ -1,16 +1,51 @@
 // Iteration #1: Find the maximum
+const maxOfTwoNumbers = (num1, num2) => (num1 > num2 ? num1 : num2);
+
+maxOfTwoNumbers(2, 4);
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words) {
+let longest = '';
+for (let i = 0; i < words.length; i++) {
+if (words[i].length > longest.length)
+longest = words[i];
+}
+return longest;
+}
+
+console.log(findLongestWord(words));
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(numbers){
+    let total=0;
+    for(let i in numbers) { 
+        total += numbers[i];
+    }
+    return total;
+}
+sumNumbers(numbers);
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+
+function averageNumbers(numbersAvg){
+    let total=0;
+    for(let i in numbersAvg) { 
+        total += numbersAvg[i]/numbersAvg.length;
+    }
+    return total;
```

Comment:
This solution seems to still give errors in the Jasmine tests.  A better approach would be to divide the `total` with the `numbersAvg.length`

```suggestion
        total += numbersAvg[i];
    }
    return total /numbersAvg.length;
```

---

This is one change, however, you have a few other things to fix. Check the Jasmine responses to see what you need to make better